extensions/incubator/src/test/java/io/opentelemetry/extension/incubator/trace/ExtendedTracerTest.java

```diff
@@ -6,112 +6,225 @@
+import io.opentelemetry.context.propagation.ContextPropagators;
+import io.opentelemetry.extension.incubator.propagation.ExtendedContextPropagators;
+import io.opentelemetry.sdk.testing.assertj.OpenTelemetryAssertions;
+import io.opentelemetry.sdk.testing.assertj.SpanDataAssert;
 import io.opentelemetry.sdk.testing.junit5.OpenTelemetryExtension;
+import io.opentelemetry.sdk.trace.data.StatusData;
+import io.opentelemetry.semconv.SemanticAttributes;
+import java.time.Instant;
+import java.util.Collections;
+import java.util.Map;
+import java.util.function.BiConsumer;
+import java.util.stream.Stream;
 import org.junit.jupiter.api.Test;
 import org.junit.jupiter.api.extension.RegisterExtension;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
 
 class ExtendedTracerTest {
+
+  interface ThrowingBiConsumer<T, U> {
+    void accept(T t, U u) throws Throwable;
+  }
+
   @RegisterExtension
   static final OpenTelemetryExtension otelTesting = OpenTelemetryExtension.create();
 
-  private final Tracer tracer = otelTesting.getOpenTelemetry().getTracer("test");
+  private final ExtendedTracer extendedTracer =
+      ExtendedTracer.create(otelTesting.getOpenTelemetry().getTracer("test"));
 
   @Test
-  void runRunnable() {
-    ExtendedTracer.create(tracer).run("testSpan", () -> Span.current().setAttribute("one", 1));
+  void wrapInSpan() {
+    assertThatIllegalStateException()
+        .isThrownBy(
+            () ->
+                extendedTracer.run(
+                    "test",
+                    () -> {
+                      // runs in span
+                      throw new IllegalStateException("ex");
+                    }));
+
+    String result =
+        extendedTracer.call(
+            "another test",
+            () -> {
+              // runs in span
+              return "result";
+            });
+    assertThat(result).isEqualTo("result");
 
     otelTesting
         .assertTraces()
+        .hasSize(2)
         .hasTracesSatisfyingExactly(
-            traceAssert ->
-                traceAssert.hasSpansSatisfyingExactly(
-                    spanDataAssert ->
-                        spanDataAssert
-                            .hasName("testSpan")
-                            .hasAttributes(Attributes.of(AttributeKey.longKey("one"), 1L))));
+            trace ->
+                trace.hasSpansSatisfyingExactly(
+                    span ->
+                        span.hasName("test")
+                            .hasStatus(StatusData.error())
+                            .hasEventsSatisfyingExactly(
+                                event ->
+                                    event
+                                        .hasName("exception")
+                                        .hasAttributesSatisfyingExactly(
+                                            OpenTelemetryAssertions.equalTo(
```

Comment:
#nit consider static import for `OpenTelemetryAssertions.equalTo` / `OpenTelemetryAssertions.satisfies` to improve readiability.